cmd/buildctl/common/common.go

```diff
@@ -21,9 +24,41 @@ func ResolveClient(c *cli.Context) (*client.Client, error) {
 		}
 		serverName = uri.Hostname()
 	}
-	caCert := c.GlobalString("tlscacert")
-	cert := c.GlobalString("tlscert")
-	key := c.GlobalString("tlskey")
+
+	var caCert string
+	var cert string
+	var key string
+
+	tlsDir := c.GlobalString("tlsdir")
+
+	if tlsDir != "" {
+		// Look for ca.pem and, if it exists, set caCert to that
+		// Look for cert.pem and, if it exists, set key to that
+		// Look for key.pem and, if it exists, set tlsDir to that
+		for _, v := range [3]string{"/ca.pem", "/cert.pem", "/key.pem"} {
+			file := filepath.Join(tlsDir + v)
+			if _, err := os.Stat(file); err == nil {
+				switch v {
+				case "/ca.pem":
+					caCert = file
+				case "/cert.pem":
+					cert = file
+				case "/key.pem":
+					key = file
+				}
+			} else {
+				return nil, errors.New(v + " not found in specified TLS directory")
```

Comment:
if error is not `os.IsNotExist()` we should not try to hide the original error. You may just wrap the error with the specific filename as well.

Reply:
Returning the error given by os.Stat(file) then. Not wrapping the error with the specific filename though because the error already returns the absolute path of the file with issues.